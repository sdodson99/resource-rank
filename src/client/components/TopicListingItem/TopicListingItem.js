import React from 'react';
import PropTypes from 'prop-types';
import styles from './TopicListingItem.module.css';
import Link from 'next/link';
import ListingItem from '../ListingItem/ListingItem';

const TopicListingItem = ({ id, name }) => {
  const resourcesLink = `/topics/${id}`;

  return (
    <div className={styles.TopicListingItem} data-testid="TopicListingItem">
      <Link href={resourcesLink}>
        <a>
          <ListingItem>
            <div>
              <div>{name}</div>
              <div>Click Me</div>
            </div>
          </ListingItem>
        </a>
      </Link>
    </div>
  );
};

TopicListingItem.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
};

TopicListingItem.defaultProps = {
  id: '',
  name: '',
};

export default TopicListingItem;
