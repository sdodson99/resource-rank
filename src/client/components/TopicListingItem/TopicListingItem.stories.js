import React from 'react';
import TopicListingItem from './TopicListingItem';

export default {
  component: TopicListingItem,
  title: 'Components/TopicListingItem',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <TopicListingItem {...args} />;

export const Default = Template.bind({});
Default.args = {
  name: 'Topic 1',
  slug: 'topic-1',
};
