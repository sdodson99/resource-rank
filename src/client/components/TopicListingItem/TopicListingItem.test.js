import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TopicListingItem from './TopicListingItem';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<TopicListingItem />', () => {
  it('should mount', () => {
    render(<TopicListingItem />);

    const topicListingItem = screen.getByTestId('TopicListingItem');

    expect(topicListingItem).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = createRenderer().render(
      <TopicListingItem name="Topic1" slug="topic1" />
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when verified', () => {
    const tree = createRenderer().render(
      <TopicListingItem name="Topic1" slug="topic1" verified={true} />
    );

    expect(tree).toMatchSnapshot();
  });
});
