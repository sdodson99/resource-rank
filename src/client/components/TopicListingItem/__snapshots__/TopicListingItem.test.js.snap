// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicListingItem /> should render correctly 1`] = `
<div
  className="TopicListingItem"
  data-testid="TopicListingItem"
>
  <Link
    href="/topics/123"
  >
    <a>
      <ListingItem>
        <div>
          <div>
            Topic1
          </div>
          <div>
            Click Me
          </div>
        </div>
      </ListingItem>
    </a>
  </Link>
</div>
`;
